### PR TITLE
Issue 17 | Sphinx Docs

### LINKED ISSUE #17: Docs: build documentation for purplecaffeine
Docs: build documentation for purplecaffeine using sphinx and github pages to deploy them

### PR DESCRIPTION
**Pull request by IceKhan13**
### Summary

Sphinx autodocs for project

### Details and comments

Screen Shot 2023-04-14 at 2 02 55 PM

Closes #17

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+# General options:
+import os
+import sys
+from pathlib import Path
+
+sys.path.append(os.path.abspath('../client'))
+
+project = "PurpleCaffeine"
+copyright = "2023"  # pylint: disable=redefined-builtin
+author = ""
```
Comment (mickahell) [inline docs/conf.py:14]: QAMP or Qiskit community ?

File: docs/conf.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+_rootdir = Path(__file__).parent.parent
+
+# The full version, including alpha/beta/rc tags
+release = "0.0.0"
+# The short X.Y version
+version = "0.0"
```
Comment (mickahell) [inline docs/conf.py:21]: latest :p ?

**Review by mickahell — APPROVED** [review]
Nice :) I just added some comments but they are details. Why did you put logging monitoring and Runtime schematics ?
And note for later: we will need to add a doc to explain how to deploy the API on our own server.

File: docs/conf.py (hunk 1 above)
Comment (IceKhan13) [reply docs/conf.py:14]: On release we will put our names :)

File: docs/conf.py (hunk 2 above)
Comment (IceKhan13) [reply docs/conf.py:21]: we will need to automate that and read data from `VERSION.txt` file later

**Comment by IceKhan13** [comment]
@mickahell agree. We will have those docs once we implement server :)

File: docs/conf.py (hunk 2 above)
Comment (mickahell) [reply docs/conf.py:21]: Even from tag, nop ?

**Pull request merged**